File: sdk/java/src/test/java/com/flagent/client/model/FlagSnapshotTest.java
package com.flagent.client.model;

import java.time.OffsetDateTime;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for FlagSnapshot
 */
class FlagSnapshotTest {

    @Test
    void testFlagSnapshot() {
        Flag flag = new Flag().id(1L).key("f1").description("d").enabled(true).dataRecordsEnabled(false);
        OffsetDateTime updatedAt = OffsetDateTime.now();
        FlagSnapshot model = new FlagSnapshot()
            .id(1L)
            .updatedBy("admin")
            .flag(flag)
            .updatedAt(updatedAt);
        Assertions.assertNotNull(model);
        Assertions.assertEquals(1L, model.getId());
        Assertions.assertEquals("admin", model.getUpdatedBy());
        Assertions.assertEquals(flag, model.getFlag());
        Assertions.assertEquals(updatedAt, model.getUpdatedAt());
    }

    @Test
    void idTest() {
        FlagSnapshot model = new FlagSnapshot().id(5L);
        Assertions.assertEquals(5L, model.getId());
    }

    @Test
    void updatedByTest() {
        FlagSnapshot model = new FlagSnapshot().updatedBy("user");
        Assertions.assertEquals("user", model.getUpdatedBy());
    }

    @Test
    void flagTest() {
        Flag flag = new Flag().id(10L).key("k").description("d").enabled(true).dataRecordsEnabled(false);
        FlagSnapshot model = new FlagSnapshot().flag(flag);
        Assertions.assertEquals(10L, model.getFlag().getId());
        Assertions.assertEquals("k", model.getFlag().getKey());
    }

    @Test
    void updatedAtTest() {
        OffsetDateTime ts = OffsetDateTime.now();
        FlagSnapshot model = new FlagSnapshot().updatedAt(ts);
        Assertions.assertEquals(ts, model.getUpdatedAt());
    }

}

File: sdk/dart/test/flag_snapshot_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for FlagSnapshot
void main() {
  final flag = Flag((b) => b
    ..id = 1
    ..key = 'snapshot_flag'
    ..description = 'Desc'
    ..enabled = true
    ..dataRecordsEnabled = false);
  final instance = FlagSnapshot((b) => b
    ..id = 100
    ..updatedBy = 'admin'
    ..flag = flag.toBuilder()
    ..updatedAt = DateTime.utc(2024, 1, 1, 12, 0, 0));

  group(FlagSnapshot, () {
    test('to test the property `id`', () async {
      expect(instance.id, equals(100));
    });

    test('to test the property `updatedBy`', () async {
      expect(instance.updatedBy, equals('admin'));
    });

    test('to test the property `flag`', () async {
      expect(instance.flag.key, equals('snapshot_flag'));
      expect(instance.flag.id, equals(1));
    });

    test('to test the property `updatedAt`', () async {
      expect(instance.updatedAt, equals(DateTime.utc(2024, 1, 1, 12, 0, 0)));
    });
  });
}
